lib/features/window/window_container.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:lantern/app.dart';
+import 'package:lantern/core/service/lantern_ffi_service.dart';
+import 'package:lantern/core/utils/common.dart';
+import 'package:window_manager/window_manager.dart';
+
+class WindowContainer extends StatefulWidget {
+  const WindowContainer(this.child, {super.key});
+
+  final Widget child;
+
+  @override
+  State<WindowContainer> createState() => _WindowContainerState();
+}
+
+class _WindowContainerState extends State<WindowContainer> with WindowListener {
+  @override
+  Widget build(BuildContext context) {
+    return widget.child;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    if (isDesktop()) {
+      windowManager.addListener(this);
+      _initializeWindow();
+      WidgetsBinding.instance.addPostFrameCallback((_) async {
+        await windowManager.setPreventClose(true);
+        await _setWindowResizable();
+      });
+    }
+  }
+
+  Future<void> _initializeWindow() async {
+    await windowManager.ensureInitialized();
+    await windowManager.setSize(const Size(360, 712));
+    await windowManager.show();
+    await windowManager.focus();
+  }
+
+  @override
+  void dispose() {
+    if (isDesktop()) windowManager.removeListener(this);
+    super.dispose();
+  }
+
+  Future<void> _setWindowResizable() async {
+    if (!Platform.isWindows) {
+      await windowManager.setResizable(false);
+      return;
+    }
+    // temporary workaround for distorted layout on Windows. The problem goes away
+    // after the window is resized.
+    // See https://github.com/leanflutter/window_manager/issues/464
+    // and https://github.com/KRTirtho/spotube/issues/1553
+    await Future<void>.delayed(const Duration(milliseconds: 100), () async {
+      windowManager.getSize().then((Size value) {
+        windowManager
+            .setSize(
+              Size(value.width + 1, value.height + 1),
+            )
+            .then((_) => setState(() => {}));
+      });
+      await windowManager.setResizable(false);
+    });
```

Comment:
I think unfortunately we still need this temporary workaround for now on Windows